node/malus/src/malus.rs

```diff
@@ -108,7 +108,10 @@ impl MalusCli {
 
 				#[cfg(not(target_os = "android"))]
 				{
-					polkadot_node_core_pvf::execute_worker_entrypoint(&cmd.socket_path);
+					polkadot_node_core_pvf::execute_worker_entrypoint(
+						&cmd.socket_path,
+						&cmd.exec_env_params_hash,
+					);
```

Comment:
Are we passing this hash here instead of through the socket, because this is not expected to change for a given worker? In that case, why not just send the actual ee params here (properly encoded), as right now we are sending the ee params over the socket for each job.

Reply:
Those are just command-line arguments :(

I would be happy to pass the whole parameter set here but I'm not sure it's a good idea to pass an object of unlimited size in SCALE-encoded form through the command line.

Reply:
Ah okay, I wasn't aware it could be unlimited size. I wonder then if there's a better way than passing it through the socket for each job, as that seems unnecessary. Maybe the params could be written to a temporary file, read once by the execute worker, and the file deleted?

Reply:
Feel free to ignore right now if this is premature optimization.

Reply:
No, I think you're right, it should be optimized. I just wanted to gather some opinions on how to do it the better way. Still hope to find a better solution than to pass it through the filesystem. Probably a handshake procedure through socket on worker startup will do. Or maybe someone will come out with even better ideas.

Also, I didn't want to rewrite the whole queue code with a single commit and was trying to utilize what is already there, and parameters passed through socket with every job were already there :)

Reply:
Reworked that part from scratch. Have a look, is it any better now?

Reply:
Yes, thanks for improving this!